src/frontends/sql/decode_tag_transform_operation_test.cc

```diff
@@ -0,0 +1,133 @@
+//-----------------------------------------------------------------------------
+// Copyright 2022 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//----------------------------------------------------------------------------
+
+#include "absl/strings/str_format.h"
+#include "absl/strings/string_view.h"
+#include "google/protobuf/text_format.h"
+#include "src/common/testing/gtest.h"
+#include "src/frontends/sql/decode.h"
+#include "src/frontends/sql/decoder_context.h"
+#include "src/frontends/sql/sql_ir.pb.h"
+#include "src/frontends/sql/testing/literal_operation_view.h"
+#include "src/frontends/sql/testing/merge_operation_view.h"
+#include "src/frontends/sql/testing/tag_transform_operation_view.h"
+#include "src/frontends/sql/testing/utils.h"
+
+namespace raksha::frontends::sql {
+
+namespace {
+
+using ir::Attribute;
+using ir::Block;
+using ir::IRContext;
+using ir::Operation;
+using ir::Storage;
+using ir::Value;
+using ir::types::TypeBase;
+using ir::value::OperationResult;
+using ir::value::StoredValue;
+using ::testing::Combine;
+using ::testing::Eq;
+using ::testing::IsEmpty;
+using ::testing::IsNull;
+using testing::LiteralOperationView;
+using testing::MergeOperationView;
+using ::testing::NotNull;
+using ::testing::Pair;
+using ::testing::ResultOf;
+using testing::TagTransformOperationView;
+using ::testing::TestWithParam;
+using ::testing::UnorderedElementsAre;
+using ::testing::UnorderedElementsAreArray;
+using testing::UnwrapDefaultOperationResult;
+using ::testing::Values;
+using ::testing::ValuesIn;
+
+class DecodeTagTransformTest
+    : public ::testing::TestWithParam<
+          std::tuple<uint64_t, absl::string_view,
+                     std::pair<absl::string_view, std::vector<uint64_t>>>> {
+ protected:
+  DecodeTagTransformTest()
+      : ir_context_(),
+        decoder_context_(ir_context_),
+        // A placeholder until we decode the value later in the ctor.
```

Comment:
It would make things easier for testing, but it might make it easier to write errors in production code. Perhaps we could add an `INVALID` or `UNINIT` value kind that errors upon any use. That, then, could be the kind of `Value` produced by the default constructor.